pkg/reconciler/taskrun/suite_test.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taskrun
+
+import (
+	"context"
+	"go/build"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/jvm-build-service/pkg/apis/jvmbuildservice/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	taskrunapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	//+kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var (
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Controller Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	By("bootstrapping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join(build.Default.GOPATH, "src", "github.com", "redhat-appstudio", "jvm-build-service", "deploy", "crds", "base"),
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "tektoncd", "pipeline@v0.33.0", "config"),
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	cfg, err := testEnv.Start()
```

Comment:
I do see you made Makefile changes similar to https://github.com/redhat-appstudio/build-service/blob/main/Makefile so perhaps you were able to run locally?  If you mentioned in slack somewhere on in one of our calls and I'm forgetting what you said, apologies.

Reply:
But otherwise, I think what you have is a good first step that we can build upon.

Reply:
@gabemontero yes, I was able to run locally against minikube and it passed

Reply:
ok that is a good start @stuartwdouglas 

now, the openshift/release stuff I noted ^^ runs against app studio deployed against an openshift cluster ... see https://github.com/redhat-appstudio/build-service/tree/main/.ci for example

while I think we would want that at some point, I have worked on projects where they launch kind or minikube from github actions for e2e testing like that as well; not knowing your experience launching / testing on openshift, vs. however painful it might be to get me to a point where I run end to end testing with this project, maybe we employ a github actions / "vanilla" k8s e2e approach for an interim period.

I'll defer to you on whether you want to include launching the e2e from github actions with this PR or a subsequent one (don't see it in https://github.com/redhat-appstudio/jvm-build-service/pull/33/files#diff-aa526adb0a3d690b8bce5b9ae75da3c3a254ebeffe4b108851b60d8746356276 currently)

long term, at first blush, it would not be a "bad thing" to have e2e tests on both vanilla k8s and openshift, though I don't see that approach being taken for the build-service or application-service repos.